components/eamxx/src/physics/rrtmgp/tests/rrtmgp_unit_tests.cpp

```diff
@@ -619,3 +619,225 @@ TEST_CASE("rrtmgp_cloud_area") {
+  // Case 1: if no clouds, everything goes to zero
+  memset(tmid, 300.0);
+  memset(pmid, 100.0);
+  memset(p_del, 10.0);
+  memset(z_del, 100.0);
+  memset(qc, 1.0);
+  memset(qi, 1.0);
+  memset(cldfrac_tot, 0.0);
+  memset(nc, 5.0);
+  memset(rel, 10.0);
+  memset(rei, 10.0);
+  // Call the function
+  scream::rrtmgp::compute_aerocom_cloudtop(
+      ncol, nlay, tmid, pmid, p_del, z_del, qc, qi, rel, rei, cldfrac_tot, nc,
+      tmid_at_cldtop, pmid_at_cldtop, cldfrac_ice_at_cldtop,
+      cldfrac_liq_at_cldtop, cldfrac_tot_at_cldtop, cdnc_at_cldtop,
+      eff_radius_qc_at_cldtop, eff_radius_qi_at_cldtop);
+
+  // Check the results
+  REQUIRE(tmid_at_cldtop.createHostCopy()(1) == 0.0);
+  REQUIRE(pmid_at_cldtop.createHostCopy()(1) == 0.0);
+  REQUIRE(cldfrac_tot_at_cldtop.createHostCopy()(1) == 0.0);
+  REQUIRE(cldfrac_liq_at_cldtop.createHostCopy()(1) == 0.0);
+  REQUIRE(cldfrac_ice_at_cldtop.createHostCopy()(1) == 0.0);
+  REQUIRE(cdnc_at_cldtop.createHostCopy()(1) == 0.0);
+  REQUIRE(eff_radius_qc_at_cldtop.createHostCopy()(1) == 0.0);
+  REQUIRE(eff_radius_qi_at_cldtop.createHostCopy()(1) == 0.0);
+
+  // Case 2: if all clouds, everything goes to 1 * its value
+  memset(cldfrac_tot, 1.0);
+  scream::rrtmgp::compute_aerocom_cloudtop(
+      ncol, nlay, tmid, pmid, p_del, z_del, qc, qi, rel, rei, cldfrac_tot, nc,
+      tmid_at_cldtop, pmid_at_cldtop, cldfrac_ice_at_cldtop,
+      cldfrac_liq_at_cldtop, cldfrac_tot_at_cldtop, cdnc_at_cldtop,
+      eff_radius_qc_at_cldtop, eff_radius_qi_at_cldtop);
+
+  REQUIRE(tmid_at_cldtop.createHostCopy()(1) == 300.0);
+  REQUIRE(pmid_at_cldtop.createHostCopy()(1) == 100.0);
+  REQUIRE(cldfrac_tot_at_cldtop.createHostCopy()(1) == 1.0);
+  REQUIRE(cldfrac_liq_at_cldtop.createHostCopy()(1) == 0.5);
+  REQUIRE(cldfrac_ice_at_cldtop.createHostCopy()(1) == 0.5);
+  REQUIRE(cdnc_at_cldtop.createHostCopy()(1) > 0.0);
+  REQUIRE(eff_radius_qc_at_cldtop.createHostCopy()(1) > 0.0);
+  REQUIRE(eff_radius_qi_at_cldtop.createHostCopy()(1) > 0.0);
+
+  // Case 3: test max overlap (if contiguous cloudy layers, then max)
+  memset(cldfrac_tot, 0.0);
+  yakl::fortran::parallel_for(
+      1, YAKL_LAMBDA(int /* dummy */) {
+        cldfrac_tot(1, 2) = 0.5;
+        cldfrac_tot(1, 3) = 0.7;
+        cldfrac_tot(1, 4) = 0.3;
+        cldfrac_tot(1, 5) = 0.2;
+      });
+  scream::rrtmgp::compute_aerocom_cloudtop(
+      ncol, nlay, tmid, pmid, p_del, z_del, qc, qi, rel, rei, cldfrac_tot, nc,
+      tmid_at_cldtop, pmid_at_cldtop, cldfrac_ice_at_cldtop,
+      cldfrac_liq_at_cldtop, cldfrac_tot_at_cldtop, cdnc_at_cldtop,
+      eff_radius_qc_at_cldtop, eff_radius_qi_at_cldtop);
+
+  REQUIRE(cldfrac_tot_at_cldtop.createHostCopy()(1) == .7);
+
+  // Case 3xtra: test max overlap
+  // This case produces >0.7 due to slight enhancement in the presence of a
+  // local minimum (0.1 is the local minimum between 0.2 and 0.4)
+  yakl::fortran::parallel_for(
+      1, YAKL_LAMBDA(int /* dummy */) {
+        cldfrac_tot(1, 5) = 0.1;
```

Comment:
Minor point, it would be clearer to include setting the other levels here too so that this test was self contained. There's nothing wrong with this though as written.